### ISSUE TITLE
[BUG] The prod 6.0.22 version of the app pulled from TestFlight on 07-July-2024 has crashed 17/22 times when opening

### ISSUE THREAD

**Issue by andreweads** [rokwire/illinois-app#4220]
The prod 6.0.22 version of the app pulled from TestFlight on 07-July-2024 has crashed 17/22 times when opening. This occurs whether opening the app from TestFlight or opening the app on its own.

**To Reproduce**

1. Pull the latest build (6.0.22) from TestFlight
2. Once installed, open the TestFlight app and tap on the Open button OR Once installed, tap on the app icon to open the app

**Expected behavior**
The app should not crash when opened.

**Smartphone (please complete the following information):**
Device: iPhone 14
OS: iOS 17.5.1
App Version: Version 6.0.22 (2122)

**Additional context**
When the app crashes, I receive a pop-up box saying "Illinois" Crashed. I have hit the Share button on that pop-up many times for these crashes and then the Submit button. Additional information for my phone, if it's helpful for those crash logs:

Model Number: MPUA3LL/A
Serial Number: LYVHVK00TN
AppleID: [aeads@illinois.edu](mailto:aeads@illinois.edu)

**Comment by andreweads** [comment]
I have launched the 6.0.22 dev and prod APKs on my Pixel 8a and haven't encountered any crashes.  I haven't launched these as many times, probably 2-4 times each, but no issues so far.

**Comment by mihail-varbanov** [comment]
It never crashed on iOS 17.3 for me. After upgrading to iOS 17.5, I was able to recreate the crash only for the TestFlight build but not at 100%. Debug or internal distribution builds never crashed for me.

The logs showed that it happens in the Firebase initialization on application startup. The research showed that this is a [known](https://stackoverflow.com/questions/78122789/crash-in-nanopb-ios-17-4-0) problem, a bug in the Firebase library, that was fixed in version 10.22.0 a few months ago.

What I did was to upgrade all Firebase libraries to their latest versions (they depend on each other so they all go together). We need to test version 6.0.24 on TestFlight to see if this has resolved the problem.

@clint156, @yoonlees ^
